codebuild/spec/buildspec_unit_coverage.yml

```diff
@@ -14,30 +14,32 @@
 version: 0.2
 
 env:
-  parameter-store:
-    CODECOV_TOKEN : codecov-upload-token
   variables:
     # CODEBUILD_ is a reserved namespace.
     CB_BIN_DIR: "./codebuild/bin"
     CC: "/usr/bin/clang"
     CXX: "/usr/bin/clang++"
 
 phases:
-  pre_build:
-    commands:
-      - |
-        if [ -d "third-party-src" ]; then
-          cd third-party-src;
-          ln -s /usr/local $CODEBUILD_SRC_DIR/third-party-src/test-deps;
-        fi
-      - ln -s /usr/local $CODEBUILD_SRC_DIR/test-deps
   build:
     on-failure: ABORT
     commands:
-      - $CB_BIN_DIR/coverage_build.sh
+    # LLVM complains about corrupt coverage information
+    # for static targets, so compile to a shared lib
+    # instead.
+      - |
+        cmake . -Bbuild \
+          -DCOVERAGE=ON \
+          -DCMAKE_PREFIX_PATH=$LIBCRYPTO_ROOT \
+          -DBUILD_SHARED_LIBS=ON
+      - cmake --build ./build -- -j $(nproc)
   post_build:
     on-failure: ABORT
     commands:
-      - LLVM_PROFILE_FILE="ut_%p.profraw" CTEST_PARALLEL_LEVEL=$(nproc) cmake --build ./build --target test ARGS="--output-on-failure -L unit"
+      - LLVM_PROFILE_FILE="ut_%8m.profraw" CTEST_PARALLEL_LEVEL=$(nproc) cmake --build ./build --target test ARGS="--output-on-failure -L unit"
       - $CB_BIN_DIR/coverage_report.sh
-      - $CB_BIN_DIR/coverage_upload.sh
+artifacts:
```

Comment:
The artifacts feature of CodeBuild is fragile- this is fine if there will _always_ be something in coverage_report, but the job will fail if there is any case where no files exist.

Reply:
That's fine, the only situation in which there wouldn't be an artifact is if something in the job has gone very wrong, in which failure would be the expected result anyways.